packages/widgets/src/health-monitoring/component.tsx

```diff
@@ -0,0 +1,343 @@
+"use client";
+
+import {
+  Avatar,
+  Box,
+  Card,
+  Center,
+  Flex,
+  Group,
+  HoverCard,
+  Indicator,
+  List,
+  Progress,
+  RingProgress,
+  Text,
+  Tooltip,
+} from "@mantine/core";
+import { useElementSize, useListState } from "@mantine/hooks";
+import {
+  IconBrain,
+  IconClock,
+  IconCpu,
+  IconCpu2,
+  IconFileReport,
+  IconInfoSquare,
+  IconServer,
+  IconTemperature,
+  IconVersions,
+} from "@tabler/icons-react";
+
+import type { HealthMonitoring } from "@homarr/integrations";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps } from "../definition";
+import { NoIntegrationSelectedError } from "../errors";
+
+export default function HealthMonitoringWidget({
+  options,
+  integrationIds,
+  serverData,
+}: WidgetComponentProps<"healthMonitoring">) {
+  if (integrationIds.length === 0) {
+    throw new NoIntegrationSelectedError();
+  }
+  const t = useI18n();
+  const [healthData] = useListState<{
+    integrationId: string;
+    healthInfo: HealthMonitoring;
+  }>(serverData?.initialData ?? []);
+
+  return (
+    <Box h="100%">
+      {healthData.map(({ integrationId, healthInfo }) => {
+        const memoryUsage = formatMemoryUsage(healthInfo.memAvailable, healthInfo.memUsed);
+        const disksData = matchFileSystemAndSmart(healthInfo.fileSystem, healthInfo.smart);
+        const { ref, width } = useElementSize();
+        const ringSize = width * 0.95;
+        const ringThickness = width / 10;
+        const progressSize = width * 0.2;
+        const formatUptime = (uptime: number) => {
```

Comment:
``uptime`` is in what unit? Suggestion: Add unit as a suffix to specify.
Example: ``ellapsedTimeSinceShutdownInMilliseconds``